pkg/sdn/plugin/egress_dns.go

```diff
@@ -0,0 +1,171 @@
+package plugin
+
+import (
+	"net"
+	"time"
+
+	"github.com/golang/glog"
+
+	osapi "github.com/openshift/origin/pkg/sdn/api"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	ktypes "k8s.io/kubernetes/pkg/types"
+)
+
+type EgressDNS struct {
+	// Holds Egress DNS entries for each policy
+	pdMap map[ktypes.UID]DNS
+}
+
+func (e *EgressDNS) Add(policy osapi.EgressNetworkPolicy) {
+	dnsInfo := DNS{}
+
+	for _, rule := range policy.Spec.Egress {
+		if len(rule.To.DNSName) > 0 {
+			if err := dnsInfo.Add(rule.To.DNSName); err != nil {
+				glog.Error(err)
+			}
+		}
+	}
+
+	if dnsInfo.Size() > 0 {
+		e.pdMap[policy.UID] = dnsInfo
+	}
+}
+
+func (e *EgressDNS) Delete(policy osapi.EgressNetworkPolicy) {
+	if _, ok := e.pdMap[policy.UID]; ok {
+		delete(e.pdMap, policy.UID)
+	}
+}
+
+func (e *EgressDNS) GetNetCIDRs(policy osapi.EgressNetworkPolicy, dnsName string) []net.IPNet {
+	var cidrs []net.IPNet
+	dnsInfo, ok := e.pdMap[policy.UID]
+	if !ok {
+		return cidrs
+	}
+
+	for name, ips := range dnsInfo.Get() {
+		if name == dnsName {
+			for _, ip := range ips {
+				if ip.To4() != nil { // IPV4 addr
+					cidrs = append(cidrs, net.IPNet{IP: ip, Mask: net.CIDRMask(32, 32)})
+				} else if ip.To16() != nil { // IPV6 addr
+					cidrs = append(cidrs, net.IPNet{IP: ip, Mask: net.CIDRMask(128, 128)})
+				} else {
+					glog.Errorf("Invalid IP: %v for policy: %s", ip, policy.Name)
+				}
+			}
+			break
+		}
+	}
+	return cidrs
+}
+
+func (plugin *OsdnNode) SyncEgressDNSPolicyRules() {
```

Comment:
I know we're already terrible about this, but I think it's confusing to keep spreading OsdnNode methods through more and more files. In this case, I think this method should go into egress_network_policy.go.